integration_tests/run_pyspark_from_build.sh

```diff
@@ -70,6 +70,30 @@ else
         # to still launch a spark application.
         TEST_PARALLEL=`nvidia-smi --query-gpu=memory.free --format=csv,noheader | awk '{if (MAX < $1){ MAX = $1}} END {print int(MAX / (2.3 * 1024)) - 1}'`
         echo "AUTO DETECTED PARALLELISM OF $TEST_PARALLEL"
+
+        # adjust TEST_PARALLEL according to cpu cores and free memory;
+        # first read from ENV; Note it's hardly to get CPU cores and real free memory in docker
+        if [[ "${PYSP_TEST_CPU_CORES}" != "" ]]; then
+          cpu_cores=${PYSP_TEST_CPU_CORES}
+        else
+          cpu_cores=$(nproc --all)
```

Comment:
Why are we passing `--all` here?  Calling `nproc` with no parameters may address @pxLi's concern about a cgroup-constrained environment.  IIUC `nproc` normally parses `/sys/fs/cgroup/cpuset/cpuset.cpus` to determine the number of CPUs available except `--all` will force it to look at all installed processors rather than available processors.

Reply:
yeah, nproc is cgroup-aware if using "--cpuset-cpus" but for not "--cpus" or "--cpu-period" or "--cpu-quota", this could also be a little tricky here if want to support all scenarios